google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporterTest.java

```diff
@@ -226,6 +225,85 @@ public void testExportingHistogramData() {
     assertThat(timeSeries.getPoints(0).getInterval().getEndTime().getNanos()).isEqualTo(endEpoch);
   }
 
+  @Test
+  public void testTimeSeriesForMetricWithGceOrGkeResource() {
+    String gceProjectId = "fake-gce-project";
+    BigtableCloudMonitoringExporter exporter =
+        new BigtableCloudMonitoringExporter(
+            projectId,
+            fakeMetricServiceClient,
+            MonitoredResource.newBuilder()
+                .setType("gce-instance")
+                .putLabels("some-gce-key", "some-gce-value")
+                .putLabels("project_id", gceProjectId)
+                .build(),
+            taskId);
+    ArgumentCaptor<CreateTimeSeriesRequest> argumentCaptor =
+        ArgumentCaptor.forClass(CreateTimeSeriesRequest.class);
+
+    UnaryCallable<CreateTimeSeriesRequest, Empty> mockCallable = mock(UnaryCallable.class);
+    when(mockMetricServiceStub.createServiceTimeSeriesCallable()).thenReturn(mockCallable);
+    ApiFuture<Empty> future = ApiFutures.immediateFuture(Empty.getDefaultInstance());
+    when(mockCallable.futureCall(argumentCaptor.capture())).thenReturn(future);
+
+    long startEpoch = 10;
+    long endEpoch = 15;
+    HistogramPointData histogramPointData =
+        ImmutableHistogramPointData.create(
+            startEpoch,
+            endEpoch,
+            Attributes.of(
+                PROJECT_ID_KEY,
+                projectId,
+                INSTANCE_ID_KEY,
+                instanceId,
+                APP_PROFILE_KEY,
+                appProfileId,
+                CLIENT_NAME_KEY,
+                clientName),
+            3d,
+            true,
+            1d, // min
+            true,
+            2d, // max
+            Arrays.asList(1.0),
+            Arrays.asList(1L, 2L));
+
+    MetricData histogramData =
+        ImmutableMetricData.createDoubleHistogram(
+            resource,
+            scope,
+            "bigtable.googleapis.com/internal/client/per_connection_error_count",
+            "description",
+            "ms",
+            ImmutableHistogramData.create(
+                AggregationTemporality.CUMULATIVE, ImmutableList.of(histogramPointData)));
+
+    exporter.export(Arrays.asList(histogramData));
+
+    CreateTimeSeriesRequest request = argumentCaptor.getValue();
+
+    assertThat(request.getName()).isEqualTo("projects/" + gceProjectId);
+    assertThat(request.getTimeSeriesList()).hasSize(1);
+
+    com.google.monitoring.v3.TimeSeries timeSeries = request.getTimeSeriesList().get(0);
+
+    assertThat(timeSeries.getResource().getLabelsMap())
+        .containsExactly("some-gce-key", "some-gce-value", "project_id", gceProjectId);
+
+    assertThat(timeSeries.getMetric().getLabelsMap()).hasSize(5);
+    assertThat(timeSeries.getMetric().getLabelsMap())
+        .containsAtLeast(PROJECT_ID_KEY.getKey(), projectId);
```

Comment:
nit: maybe we can rename projectId to bigtableProjectId to stay consistent with the GCE one and the main class.